File: PASCAL/Cap04/c04ex04.pas
program c04ex04;

var
  N1, N2, N3, N4, NE: Real;
  MD1, MD2: Real;

begin
  Write('Entre a 1a. nota .......: ');
  ReadLn(N1);

  Write('Entre a 2a. nota .......: ');
  ReadLn(N2);

  Write('Entre a 3a. nota .......: ');
  ReadLn(N3);

  Write('Entre a 4a. nota .......: ');
  ReadLn(N4);
  WriteLn;

  MD1 := (N1 + N2 + N3 + N4) / 4;

  if (MD1 >= 7) then
    begin
      WriteLn('Aprovado');
      WriteLn('Media: ', MD1:0:2);
    end
  else
    begin
      Write('Entre a nota de exame ..: ');
      ReadLn(NE);
      WriteLn;
      
      MD2 := (MD1 + NE) / 2;
      
      if (MD2 >= 5) then
        WriteLn('Aprovado em exame')
      else
        WriteLn('Reprovado');
      WriteLn('Media: ', MD2:0:2);
    end;

  WriteLn;
  Write('Tecle <Enter> para encerrar... ');
  ReadLn;
end.

File: LUA/Cap04/c04ex04.lua
-- c04ex04.lua

io.write("Entre a 1a. nota .......: ")
n1 = tonumber(io.read())

io.write("Entre a 2a. nota .......: ")
n2 = tonumber(io.read())

io.write("Entre a 3a. nota .......: ")
n3 = tonumber(io.read())

io.write("Entre a 4a. nota .......: ")
n4 = tonumber(io.read())

md1 = (n1 + n2 + n3 + n4) / 4

if (md1 >= 7) then
  print()
  print("Aprovado")
  print("Media: " .. string.format("%.2f", md1))
else
  io.write("Entre a nota de exame ..: ")
  ne = tonumber(io.read())
  md2 = (md1 + ne) / 2
  if (md2 >= 5) then
    print()
    print("Aprovado em exame")
  else
    print()
    print("Reprovado")
  end
  print("Media: " .. string.format("%.2f", md2))
end

print()
io.write("Tecle <Enter> para encerrar... ")
io.read()
